File: problems/0001 - 0400/0367/0367.py
class Solution:
    def isPerfectSquare(self, num: int) -> bool:
        start = 1
        end = num
        while start <= end:
            mid = start + (end - start) // 2
            mid2 = mid ** 2
            if mid2 == num:
                return True
            elif mid2 < num:
                start = mid + 1
            else:
                end = mid - 1
        return False

File: problems/0001 - 0400/0367/0367.cpp
class Solution {
public:
    bool isPerfectSquare(int num) {
        unsigned int start = 1;
        unsigned int end = num;
        while (start <= end) {
            unsigned int mid = start + (end - start) / 2;
            unsigned long long mid2 = (unsigned long long)mid * mid;
            if (mid2 == num) {
                return true;
            }
            else if (mid2 > num) {
                end = mid - 1;
            }
            else {
                start = mid + 1;
            }
        }
        return false;
    }
};
